Library/Homebrew/cmd/outdated.rb

```diff
@@ -17,9 +17,21 @@ def outdated
 
   def outdated_brews(formulae)
     formulae.map do |f|
-      versions = f.rack.subdirs.map { |d| Keg.new(d).version }.sort!
-        if versions.all? { |version| f.pkg_version > version }
-        yield f, versions if block_given?
+      all_versions = []
+      same_tap_versions = []
+      f.rack.subdirs.each do |dir|
+        keg = Keg.new dir
+        version = keg.version
+        all_versions << version
+
+        tap = Tab.for_keg(keg).tapped_from
+        if f.tap == tap || f.version == version
```

Comment:
Two questions: what does `f.version == version` do here, and should it be `f.pkg_version`?